src/comments.js

```diff
@@ -412,19 +416,27 @@ function printComments(path, print, options) {
             parent
           )
         );
+      } else {
+        // Those two node types print their dangling comments themselves
+        if (value.type !== "Program" && value.type !== "BlockStatement") {
+          trailingParts.push(
+            printDanglingComment(
+              commentPath,
+              print,
+              options
+            )
+          );
+        }
```

Comment:
We should definitely do that but I'm a bit hesitant to not have an automatic fallback. If we miss one place, we're going to drop comments, which is really bad.